drop-in/src/main/java/com/adyen/checkout/dropin/internal/ui/PreselectedStoredPaymentMethodFragment.kt

```diff
@@ -234,6 +212,40 @@ internal class PreselectedStoredPaymentMethodFragment : DropInBottomSheetDialogF
             .show()
     }
 
+    private fun showConfirmationPopup(paymentMethodName: String, storedPaymentMethodModel: StoredPaymentMethodModel) {
+        val dialog = AlertDialog.Builder(requireContext())
+            .setTitle(
+                String.format(
+                    resources.getString(R.string.checkout_stored_payment_confirmation_message),
+                    paymentMethodName,
+                ),
+            )
+            .setNegativeButton(R.string.checkout_stored_payment_confirmation_cancel_button) { dialog, _ ->
+                dialog.dismiss()
+            }
+            .setPositiveButton(
+                PayButtonFormatter.getPayButtonText(
+                    amount = dropInViewModel.dropInParams.amount,
+                    locale = dropInViewModel.dropInParams.shopperLocale,
+                    localizedContext = requireContext(),
+                ),
+            ) { dialog, _ ->
+                dialog.dismiss()
+                storedPaymentViewModel.onConfirmed()
+            }
+
+        val message = when (storedPaymentMethodModel) {
+            is StoredCardModel ->
+                requireActivity().getString(R.string.last_four_digits_format, storedPaymentMethodModel.lastFour)
+            is GenericStoredModel -> null
+            is StoredPayByBankUSModel -> storedPaymentMethodModel.name
+            is StoredACHDirectDebitModel ->
+                requireActivity().getString(R.string.last_four_digits_format, storedPaymentMethodModel.lastFour,)
```

Comment:
Is it possible to get this from the `storedPaymentMethodItem`? I see that we already set the title there

Reply:
It doesn't match perfectly because `GenericStoredModel` doesn't show a pop up but has a title. However I added a `popUpMessage` field to `storedPaymentMethodItem`, that saves us some duplicate code.